spring-data-microservice-transaction-sample/README.md

```diff
@@ -0,0 +1,548 @@
+
+`executeTwoPcTransaction()` API, called on Order Service, automatically performs preparations, validations and commits of both the local Order Service and the remote Customer Serivice. These steps are executed sequentially after the above CRUD operations successfully finish. The implementations to invoke `prepare`, `validate` and `commit` gRPC endpoints of Customer Service need to be passed as parameters to the API (the code is [here](order-service/src/main/java/sample/order/OrderService.java#L134-L141)):
+
+```java
+Collections.singletonList(
+  RemotePrepareCommitPhaseOperations.createSerializable(
+    this::callPrepareEndpoint,
+    this::callValidateEndpoint,
+    this::callCommitEndpoint,
+    this::callRollbackEndpoint
+  )
+)
+```
+
+![Sequence Diagram of High Level 2PC API](images/seq-diagram-high-level-2pc-api.png)
+
+In the `prepare` endpoint of Customer Service, it resumes and prepares the transaction (the code is [here](customer-service/src/main/java/sample/customer/CustomerService.java#L122-L126)):
+
+```java
+execTwoPcOperation(request.getTransactionId(), false, responseObserver, "Payment", () -> {
+  // Prepare the transaction
+  customerRepository.prepare();
+});
+```
+
+The transaction is resumed in `execTwoPcOperation()` as shown above.
+
+
+In the `validate` endpoint of Customer Service, the microservice resumes and validates the transaction (the code is [here](customer-service/src/main/java/sample/customer/CustomerService.java#L131-L135)):
+
+```java
+execTwoPcOperation(request.getTransactionId(), false, responseObserver, "Validate", () -> {
+  // Validate the transaction
+  customerRepository.validate();
+  return Empty.getDefaultInstance();
+});
+```
+
+In the `commit` endpoint of Customer Service, the microservice resumes and commits the transaction (the code is [here](customer-service/src/main/java/sample/customer/CustomerService.java#L140-L144)):
+
+```java
+execTwoPcOperation(request.getTransactionId(), false, responseObserver, "Commit", () -> {
+  // Commit the transaction
+  customerRepository.commit();
+  return Empty.getDefaultInstance();
+});
+```
+
+### Error handling
+
+If an error occurs during the transaction, the transaction will be automatically rolled back by using `executeTwoPcTransaction()`. The implementation to invoke the `rollback` gRPC endpoint of Customer Service also needs to be passed as a parameter to the API with other ones (the code is [here](order-service/src/main/java/sample/order/OrderService.java#L134-L141)):
+
+```java
+Collections.singletonList(
+  RemotePrepareCommitPhaseOperations.createSerializable(
+    this::callPrepareEndpoint,
+    this::callValidateEndpoint,
+    this::callCommitEndpoint,
+    this::callRollbackEndpoint
+  )
+)
+```
+
+In the `rollback` endpoint of Customer Service, the microservice resumes and rolls back the transaction (the code is [here](customer-service/src/main/java/sample/customer/CustomerService.java#L149-L153)):
```

Comment:
```suggestion
In the `rollback` endpoint of the Customer Service, the microservice resumes and rolls back the transaction (the code is [here](customer-service/src/main/java/sample/customer/CustomerService.java#L149-L153)):
```

Reply:
Applying this change in https://github.com/scalar-labs/scalardb-samples/pull/40